ergotree-interpreter/src/sigma_protocol/fiat_shamir.rs

```diff
@@ -17,19 +17,35 @@ use thiserror::Error;
 #[cfg(feature = "arbitrary")]
 use proptest_derive::Arbitrary;
 
+use super::proof_tree::ProofTree;
+use super::GROUP_SIZE;
+use super::SOUNDNESS_BYTES;
+
 /// Hash type for Fiat-Shamir hash function (24-bytes)
 #[cfg_attr(feature = "arbitrary", derive(Arbitrary))]
 #[derive(PartialEq, Eq, Debug, Clone)]
 pub struct FiatShamirHash(pub Box<[u8; SOUNDNESS_BYTES]>);
 
+impl FiatShamirHash {
+    pub fn secure_random() -> Self {
```

Comment:
Why this method is in FiatShamirHash? Should be in something like CryptoUtils

Reply:
Thanks for catching this! I moved a "real" random bytes generator `secure_random_bytes` function to the new `crypto_utils` module. This is just a "wrapper" for `secure_random_bytes` to ensure the size of the generated byte array is correct. It produces an instance of this type (`FiatShamirHash`) so think of it as a constructor for this type.